torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1200,6 +1200,39 @@ def sample_inputs_svd(op_info, device, dtype, requires_grad=False):
 def sample_inputs_linalg_svd(op_info, device, dtype, requires_grad=False):
     return _sample_inputs_svd(op_info, device, dtype, requires_grad, is_linalg_svd=True)
 
+def sample_inputs_eig(op_info, device, dtype, requires_grad=False):
+    eigvecs = make_tensor((S, S), device=device, dtype=dtype,
+                          low=None, high=None)
+    eigvals = make_tensor((S,), device=device, dtype=dtype,
+                          low=None, high=None)
+    input = (eigvecs * eigvals.unsqueeze(-2)) @ eigvecs.inverse()
```

Comment:
Why is the input generated from eigvecs and eigvals? Is it better than generating the input directly just with
```python
input = make_tensor((S, S), device=device, dtype=dtype, low=None, high=None)
```

Reply:
Sorry, I do not get the question. This way we generate diagonazible inputs with non-complex eigenvalues for non-complex dtypes.

Reply:
Alright, I think having a comment about that here in the test would be useful.